core/src/serde.rs

```diff
@@ -177,28 +195,29 @@ impl From<(Addresses, NetworkPrefix)> for AddressesSerde {
     }
 }
 
-impl TryFrom<(AddressesSerde, NetworkPrefix)> for Addresses {
+impl TryFrom<AddressesSerde> for (Addresses, NetworkPrefix) {
     type Error = AddressEncoderError;
-    fn try_from(t: (AddressesSerde, NetworkPrefix)) -> Result<Self, Self::Error> {
-        let addresses = t.0;
-        let prefix = t.1;
-        let encoder = AddressEncoder::new(prefix);
-        Ok(Addresses {
-            address_for_oracle_tokens: encoder
-                .parse_address_from_str(&addresses.address_for_oracle_tokens)?,
-            wallet_address_for_chain_transaction: encoder
-                .parse_address_from_str(&addresses.wallet_address_for_chain_transaction)?,
-        })
+    fn try_from(addresses: AddressesSerde) -> Result<Self, Self::Error> {
+        let oracle_token_network_addr = AddressEncoder::unchecked_parse_network_address_from_str(
+            &addresses.address_for_oracle_tokens,
+        )?;
+        let prefix = oracle_token_network_addr.network();
+        let wallet_address_for_chain_transaction = AddressEncoder::new(prefix)
+            .parse_address_from_str(&addresses.wallet_address_for_chain_transaction)?;
+        Ok((
+            Addresses {
+                address_for_oracle_tokens: oracle_token_network_addr.address(),
+                wallet_address_for_chain_transaction,
+            },
+            prefix,
+        ))
     }
 }
 
-impl From<BootstrapConfig> for BootstrapConfigSerde {
-    fn from(c: BootstrapConfig) -> Self {
-        let prefix = if c.on_mainnet {
-            NetworkPrefix::Mainnet
-        } else {
-            NetworkPrefix::Testnet
-        };
+impl From<(BootstrapConfig, NetworkPrefix)> for BootstrapConfigSerde {
```

Comment:
Seems like the network prefix is only needed here for config template(example) generation. We probably can hardcode mainnet there since it must be changed by the user anyway. This should allow ditching `Command::Bootstrap::testnet` field.